ui/mirage/services/project.ts

```diff
@@ -23,3 +23,26 @@ export function get(schema: any, { requestBody }: Request): Response {
 
   return this.serialize(resp, 'application');
 }
+
+export function update(schema: any, { requestBody }: Request): Response {
+  let requestMsg = decode(UpsertProjectRequest, requestBody);
+  let name = requestMsg.getProject().getName();
+  let variablesList = requestMsg
+    .getProject()
+    .getVariablesList()
+    .map((v) => v.toObject());
+  let model = schema.projects.findBy({ name });
+
+  let newVars = [];
+  variablesList.forEach((v) => {
+    newVars.push(schema.variables.create(v));
+  });
+
+  model.variables.models = newVars;
```

Comment:
Mirage has a nice convenience for this:

```suggestion
  variablesList.forEach((v) => model.createVariable(v));
```

[See the mirage guides for documentation](https://miragejs.com/docs/main-concepts/relationships/#hasmany)